share/cluster/consul.go

```diff
@@ -1349,6 +1372,9 @@ func (m *consulMethod) RegisterExistingWatchers() {
 }
 
 func (m *consulMethod) RegisterWatcherMonitor(failFunc func() bool, recoverFunc func()) {
+	watchPlansLock.RLock()
+	defer watchPlansLock.RUnlock()
+
```

Comment:
Move the lock after the debug log. Thus, we can learn what is the last function holds the lock.